src/ai/ai.cpp

```diff
@@ -160,8 +163,11 @@ void form_alliances(sys::state& state) {
 					else
 						return a.index() > b.index();
 				});
-				nations::make_alliance(state, n, alliance_targets[0]);
-
+				if(state.world.nation_get_is_player_controlled(alliance_targets[0])) {
+					command::execute_ask_for_alliance(state, n, alliance_targets[0]);
```

Comment:
the alliance target list should never have human-controlled nations in it

Reply:
AI can now ask player for alliance, which will make life infinitely easier for most players
and given that its not as spammy as popups, it will be better for people using the topbar thing above